source/Halibut/ServiceModel/IPendingRequestQueue.cs

```diff
@@ -7,6 +8,6 @@ public interface IPendingRequestQueue
     {
         bool IsEmpty { get; }
         void ApplyResponse(ResponseMessage response);
-        RequestMessage Dequeue();
+        Task<RequestMessage> Dequeue();
```

Comment:
Wonder if we should do this as a non-breaking additive change by adding `Task<RequestMessage> DequeueAsync();`?